core/src/apps/benfen/tx_parser.py

```
@@ -0,0 +1,431 @@
+def parse_transaction_expiration(parser) -> dict | None:
+    expiration_type = parser.read_u8()
+    if expiration_type == 0:
+        return {"type": "None", "value": None}
+    elif expiration_type == 1:
+        epoch = parser.read_u64()
+        return {"type": "Epoch", "value": epoch}
+    else:
+        return None
+
+
+def parse_transaction(hex_data) -> dict | None:
```

Ruff (0.8.2)

Missing type annotation for function argument `hex_data`

(ANN001)